gravity-forms/gw-choice-counter.php

```diff
@@ -107,18 +108,28 @@ function output_script() {
 						return Boolean( $field.find( 'input[type="checkbox"]' ).length );
 					}
 
-					self.updateChoiceCount = function( formId, choiceFieldIds, countFieldId ) {
+					self.updateChoiceCount = function( formId, choiceFieldIds, countFieldId, values ) {
 
 						var countField = $( '#input_' + formId + '_' + countFieldId ),
 							count      = 0;
 
 						for ( var i = 0; i < choiceFieldIds.length; i++ ) {
 
 							var $choiceField = $( '#input_' + formId + '_' + choiceFieldIds[ i ] );
-							if ( self.isCheckboxField( $choiceField ) ) {
-								count += $choiceField.find( 'input[type="checkbox"]:checked' ).not(' #choice_' + choiceFieldIds[ i ] + '_select_all').length;
+							if ( values.length == 0) {
+								if ( self.isCheckboxField( $choiceField ) ) {
+									count += $choiceField.find( 'input[type="checkbox"]:checked' ).not(' #choice_' + choiceFieldIds[ i ] + '_select_all').length;
+								} else {
+									count += $choiceField.find( 'option:selected' ).length;
+								}
 							} else {
-								count += $choiceField.find( 'option:selected' ).length;
+								var selectedValues = [];
+								$choiceField.find( 'input[type="checkbox"]:checked' ).each( function( k, $selectedChoice ) {
+									selectedValues.push( $selectedChoice.value );
+								});
+								values.forEach( function( val ) {
+									count += selectedValues.indexOf( val ) >= 0;
+								});
 							}
```

Comment:
Can you add some inline comments here of what's going on?

Reply:
Added